evalml/automl/automl_search.py

```diff
@@ -667,6 +676,9 @@ def __init__(
             pipeline_params=parameters,
             custom_hyperparameters=custom_hyperparameters,
         )
+        if self.verbose:
+            logger.handlers.pop()
+            logger.setLevel(logging.WARNING)
```

Comment:
Why do this at the end of init? Why not put it with the other logger init code at the top of the init method?

Reply:
Also, calling `pop` implies we're assuming the last handler in the list on `logger.handlers` is one that we added and has to be deleted. What's the reason for doing so here?

Reply:
I referenced this in my above comment, but it's to ensure we undo our logging configuration as soon as we no longer need it! (Briefly discussed in the design doc [here](https://alteryx.atlassian.net/wiki/spaces/PS/pages/972456879/Simplify+EvalML+logging?focusedCommentId=983695807#comment-983695807))